src/creature.cpp

```diff
@@ -1432,6 +1432,18 @@ void Creature::process_effects()
             if( e.get_intensity() != prev_int && e.get_duration() > 0_turns ) {
                 on_effect_int_change( e.get_id(), e.get_intensity(), e.get_bp() );
             }
+
+            const bool reduced = resists_effect( e );
+            if( e.kill_roll( reduced ) ) {
+                add_msg_if_player( m_bad, e.get_death_message() );
+                if( is_player() ) {
+                    /*
+                            event_type aaa = e.death_event();
+                            get_event_bus().send<aaa>( as_player()->getID(), elem.first );
+                    */
```

Comment:
At least for the initial implementation I suggest you require that the `death_event` be of a type with a single field `character` of type `character_id` (these are all the events whose `event_spec` specialization inherits from `event_spec_character`; there are quite a few, including `throws_up`).  Then you can construct the event at runtime here.

Just be sure to add something to the verify step that checks the chosen event type is of the required form.  You can use `event::get_fields` to determine the fields for a particular type known at runtime.